### PR TITLE
docker compose implemented to athenaFW

### PR DESCRIPTION
**Pull request by brkfun**
Hey guys today i tried myself to create a docker compose speciality to your project. It was an easy solution but if someone wants to try this first has to wget altv-server from cdn like below

`wget -q "https://cdn.altv.mp/server/release/x64_linux/altv-server"`

### CODE REVIEW

File: server.dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM mcr.microsoft.com/dotnet/sdk:6.0
+RUN mkdir /app
+WORKDIR /app
+COPY . .
+RUN curl -sL https://deb.nodesource.com/setup_18.x | bash -
+RUN apt-get install -y nodejs
+RUN apt update && apt install -y wget bash libatomic1 procps
+RUN wget -q "https://cdn.altv.mp/server/release/x64_linux/altv-server"
```
Comment (Stuyk) [inline server.dockerfile:8]: Pretty sure you should be able to do `npm run update` here?

Comment (brkfun) [reply server.dockerfile:8]: this is added by next commit 👍

File: AthenaConfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "[?] A URI with authentication to connect to a MongoDB Database": "TYPE IS STRING",
     "[?] ex: mongodb://username:password@127.0.0.1:27017": "TYPE IS STRING",
-    "MONGO_URL": null,
+    "MONGO_URL": "mongodb://root:root@mongodb:27017",
```
[the remark this answers is not in the document]
Comment (brkfun) [reply AthenaConfig.json]: this is resolved by next commit 👍

File: server.dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
```
Comment (Stuyk) [inline server.dockerfile:8]:
```
RUN wget -q "https://cdn.altv.mp/server/release/x64_linux/altv-server"
RUN wget -q "https://cdn.altv.mp/js-module/release/x64_linux/modules/js-module/libjs-module.so" -P modules/js-module/
RUN wget -q "https://cdn.altv.mp/js-module/release/x64_linux/modules/js-module/libnode.so.102" -P modules/js-module/
```

These are all obtained by `npm run update`

```
chmod 777 altv-server
```

Should be moved to after `npm run update`.

**Comment by Stuyk** [comment]
Some other thoughts on this current Docker setup.

As Athena is meant to be changed and manipulated over time.

I don't believe the proper entry point is `npm run dev` it instead should be simply allowing the docker user to access the image and run whichever environment they need to run. Such as `npm run linux`, `npm run dev`, etc.

Different environments have different meanings and `dev` is meant for development with the server infrastructure usually in a windows environment. Docker pretty much makes this impossible because any changes you make are no longer saved.

There's no perfect solution for a docker based environment.

**Things that need to be considered:**

* Obtaining code from a private repository
* Obtaining authorization to a private repository
* Creating the image based on secrets passed to the docker image
* Keeping those secrets actually secret by using BUILDKIT and other forms of safety measures
* Setting up 'volumes' to keep code safe while working on it and without losing it between docker image builds

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+
+version: "3.3"
+
+services:
+  mongodb:
+    ports:
+      - "27017:27017"
+    container_name: altv-mongo
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=root
```
Comment (Stuyk) [inline docker-compose.yml:10]: Shouldn't be hard coding these into the environment. It's a security risk and the secrets get baked into the image layers.

See: https://earthly.dev/blog/docker-secrets/

Comment (brkfun) [reply docker-compose.yml:10]: no that's not hard. But as i want to mention; i need to refine this pr i think. I do this in 10 mins or so. Let me deep dive at this weekend.

File: server.dockerfile (hunk 3 above)
Comment (brkfun) [reply server.dockerfile:8]: actually i didn't aware of this command sorry about that :D npm run update was a good command for this docker image man. :D i'll fix this up today.

**Comment by brkfun** [comment]
> 
As you mentioned npm run dev is not working. Actually i need to understand how this framework and altv works. I'll try my best to just understand this project for today.

**Pull request closed without merging**